### DISCUSSION TITLE
Where is Rust used in the project?

### DISCUSSION THREAD

**Discussion by kingmo888** [nautechsystems/nautilus_trader#821] [General]
Where is Rust used in the project?
Recently, I was looking for a transaction framework written with Rust, and finally found Nautilus trader,
To this end, envs is specially used to create a py310 environment to install nautilus trader,
At the same time, I used cargo build -- release to compile nautilus_ Core, the dll file generated through query, has not been used in the whole computer.
I checked all the documents and found no relevant parts of Rust.
So I have a question. In what way does Rust participate in the project?

**Comment by cjdsellers** [comment]
Hi @kingmo888 

Thanks for your interest in the project. So you'll find all Rust source code in the `nautilus_core` directory, which you seem to have found because you ran cargo build on it.

So the way the Rust libraries are used is `cargo` will generate static libraries, then when the C extension modules are being compiled they are statically linked. You might have a slightly older version where we still produced the dynamic libraries, the build config was changed recently because they aren't used.

The Rust libraries currently provide some core value objects, some of the domain model, core components such as clocks, timers and loggers. There's an order book which will be integrated soon, and also some data loading capability for parquet (integration almost done).

https://docs.nautilustrader.io/user_guide/architecture.html#code-structure
